src/app/src/components/FacilityListItemsTableSearch.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from 'react';
+import { func, string } from 'prop-types';
+
+import CircularProgress from '@material-ui/core/CircularProgress';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import TextField from '@material-ui/core/TextField';
+import Search from '@material-ui/icons/Search';
+import Close from '@material-ui/icons/Close';
+
+const KEYCODE_ENTER = 13;
+const KEYCODE_ESCAPE = 27;
+
+export default function FacilityListItemsTableSearch({
+    currentSearchTerm,
+    onSearch,
+}) {
+    const [isSearching, setSearching] = useState(false);
+    const [isCurrent, setCurrent] = useState(true);
+    const [searchText, setSearchText] = useState(currentSearchTerm);
```

Comment:
Question: since this last `useState` is set from a prop, what happens if the prop's value changes? I assume the initial value is captured on mounting and doesn't automatically update thereafter but I'm not entirely sure.

I think this may be a good case for the useEffect hook: essentially if `currentSearchTerm` ever changes from above I think we want to ensure that `setSearchText(currentSearchTerm)` is called. I think it'd look like:

```
const [searchText, setSearchText] = useState('');

useEffect(() => {
    setSearchTerm(currentSearchTerm);
}, [currentSearchTerm, setSearchText]);
```

The quirk here generally seems to be that we're storing `currentSearchTerm` somewhere else (either Redux or the querystring param?) but then scoping the changes to the actual text input only here for some part of the lifecycle.

Reply:
My understanding was that if the prop value changes then the whole thing is re-rendered.

Reply:
This also may circumvent needing to have `isCurrent` around? It's not clear from the var name what `isCurrent` represents -- although I think it may be a bool indicating whether the search input state is the same as `currentSearchTerm`.

Reply:
> My understanding was that if the prop value changes then the whole thing is re-rendered.

We should check this to make sure. I was thinking that `useState` would only be called once, on initially mounting the component -- otherwise the state wouldn't be durable across renders.

Reply:
Found this: https://stackoverflow.com/a/54866051

> The argument passed to useState is the initial state much like setting state in constructor for a class component and isn't used to update the state on re-render

Reply:
Yes I'm just finding out the same thing. Alright, will switch to another hook.